packages/frontend/components/Sidebar/NUPathSection.tsx

```diff
@@ -0,0 +1,283 @@
+import { NUPathEnum, ScheduleCourse2 } from "@graduate/common";
+import { SidebarValidationStatus } from "./Sidebar";
+import { useState } from "react";
+import { Box, Flex, Spinner, Text } from "@chakra-ui/react";
+import { HelperToolTip } from "../Help";
+import {
+  CheckIcon,
+  ChevronDownIcon,
+  ChevronUpIcon,
+  SmallCloseIcon,
+} from "@chakra-ui/icons";
+
+interface NUPathSectionProps {
+  coursesTaken: ScheduleCourse2<unknown>[];
+  dndIdPrefix: string;
+  loading?: boolean;
+}
+const nuPathDisplayAndAbbr: [
```

Comment:
Instead of an array, can we switch this to an Object where the key is `NUPathEnum`? So it would be a `Record<NUPathEnum, string>`

Reply:
Not doing this to make it easy to iterate over in the component